src/main/java/date/time/DateTimeParser.java

```diff
@@ -0,0 +1,141 @@
+
+/**
+ * This class encapsulates a Date Time Parser. It contains methods to parse inputs into a
+ * LocalDateTime object, as well as a checker for the format.
+ *
+ * @author Ong Han Yang
+ */
+public final class DateTimeParser {
+    /** Reusable Invalid Input Exception for when the input command has an invalid date format */
+    public static final InvalidInputException INVALID_DATE_TIME_FORMAT =
+            new InvalidInputException("The date/time format is wrong! Please enter in this format: yyyy-mm-dd hh:mm");
+
+    /**
+     * Parses a string input in the form yyyy-mm-dd hh:mm, where dd is the day, mm is the month,
+     * yyyy is the year and hh:mm is the 24 hour time, to produce a LocalDateTime object
+     * with the provided date and time.
+     *
+     * Does not check for invalid date/time.
+     *
+     * @param input the string input in the form yyyy-mm-dd xx:xx .
+     * @return the LocalDateTime corresponding to the input.
+     */
+    public static LocalDateTime parse(String input) {
+        String[] split = input.split(" ", 2);
+        int[] date = Arrays.stream(split[0]
+                .split("-", 3))
+                .mapToInt(Integer::parseInt)
+                .toArray();
+        int[] time = Arrays.stream(split[1]
+                .split(":", 2))
+                .mapToInt(Integer::parseInt)
+                .toArray();
+        return LocalDateTime.of(
+                date[0],
+                date[1],
+                date[2],
+                time[0],
+                time[1]);
+    }
+
+    /**
+     * Parses a string in the format of {@code hh:mm} to produce a LocalTime object with the specified
+     * hour and minute.
+     *
+     * Does not check for invalid time.
+     *
+     * @param input a String in the form of {@code hh:mm}.
+     * @return a LocalTime object with the specified time.
+     */
+    public static LocalTime parseTime(String input) {
+        String[] split = input.split(":", 2);
+        return LocalTime.of(Integer.parseInt(split[0]), Integer.parseInt(split[1]));
+    }
+
+    /**
+     * Checks if a given string input has a valid format to be converted into
+     * a LocalDateTime using the DateTimeParser::parse method.
+     *
+     * @param input The string to be checked if it follows yyyy-mm-dd hh:mm .
+     * @return true if there are no exceptions.
+     * @throws InvalidInputException when a given format is wrong.
+     */
+    public static boolean checkValidFormat(String input) throws InvalidInputException {
+        if (input.length() != 16) {
+            throw INVALID_DATE_TIME_FORMAT;
+        }
+        // check for separators and numbers
+        // 0123456789012345
+        // yyyy-mm-dd xx:xx
+        // if wrong format, return false
+        // Any "magic numbers" seen from this point onwards refer to the indexes of the specific
+        // string to check, as shown above in line 80.
+        for (int i = 0; i < input.length(); i++) {
```

Comment:
Hmmm, what do you mean?
I thought about maybe abstracting each part away into individual helper functions/methods but I'm not too sure how I would use a switch statement here

Reply:
I was thinking of using switch(i) to check the value of the index instead of multiple if/else statements. This will also help to reduce the amount of nesting and make the code look nicer. But on a second look I guess the amount of nesting wasn't that bad. 

I don't think it is necessary as this check is a pretty simple check. Rather than that, 1 suggestion I have(but may not be the most optimal) would be to use String::split and then doing the necessary checks by 1) checking the appropriate lengths of each split, and 2) checking if the splits that are supposed to be digits are digits. The con of this is that it might lengthen your code, but it avoids magic numbers. Let me know what you think.

Reply:
Hmm, I'm not particularly fond of using String::split as it relies heavily on the presence of the regex given. This is especially the case where I would need to split multiple times, where I need to check for the length multiple times in addition to creating many objects and eventually enumerating through all the non-separators/dividers to ensure that they are all numbers/digits.

I do agree with how it will remove the magic numbers though, but I think I will keep to this implementation for now? Since I did explain the magic numbers and they aren't too hard to understand